jaxrs-api/src/main/java/jakarta/ws/rs/BadGatewayException.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2012, 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package jakarta.ws.rs;
+
+import jakarta.ws.rs.core.Response;
+
+/**
+ * A runtime exception indicating that the server, while acting as a gateway or proxy,
+ * received an invalid response from the upstream server it accessed in attempting to fulfill the request.
+ *
+ * @author Jeyvison Nascimento
+ * @since 2.0
+ */
+public class BadGatewayException extends ServerErrorException {
+
+    private static final long serialVersionUID = 7264647684649480265L;
```

Comment:
@eclipse-ee4j/eclipse-jaxrs Do we really want to use `serialVersionUID`s? It implies that if we change the class we MUST modify the UID *always*.

Reply:
It's used in other exceptions classes but i get the point if it's really needed.

Reply:
@eclipse-ee4j/eclipse-jaxrs Opinions?

Reply:
Well, `Throwable` implements `Serializable`. And the API docs recommends defining the `serialVersionUID` manually. I don't think that the exceptions will be changed in a binary incompatible way very often, so I'm fine with including the `serialVersionUID` and adapting it if required.

Reply:
I don't see a problem `serialVersionUID ` keeping them too. as @chkal said, we can adapt it later if we need to.

Reply:
Dumb question, why not simply using `1L;`? I know IDEs create these fancy numbers, but what are these scary codes good for in the sense of "better than 1L"?

Reply:
According to the [serialization spec](https://docs.oracle.com/javase/6/docs/platform/serialization/spec/version.html#6519), this is to guard against classes with the same name being identified as versions of the same class.

Reply:
@jeyvison I know what a SUID is good for. Actually I do not see why a SUID of 1L wouldn't suffice. It identifies the stream version as good as any other number.

Reply:
Then I'm ok with just 1L. It's cleaner.